File: frontend/apps/admin/src/rpc/service/organization.ts
import type { OrganizationService } from '#/rpc/api/admin/service/v1/i_organization.pb';
import type { Empty } from '#/rpc/api/google/protobuf/empty.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';
import type {
  CreateOrganizationRequest,
  DeleteOrganizationRequest,
  GetOrganizationRequest,
  ListOrganizationResponse,
  Organization,
  UpdateOrganizationRequest,
} from '#/rpc/api/user/service/v1/organization.pb';

import { requestClient } from '#/rpc/request';

/** 组织管理服务 */
class OrganizationServiceImpl implements OrganizationService {
  async CreateOrganization(request: CreateOrganizationRequest): Promise<Empty> {
    return await requestClient.post<Empty>('/organizations', request);
  }

  async DeleteOrganization(request: DeleteOrganizationRequest): Promise<Empty> {
    return await requestClient.delete<Empty>(`/organizations/${request.id}`);
  }

  async GetOrganization(
    request: GetOrganizationRequest,
  ): Promise<Organization> {
    return await requestClient.get<Organization>(
      `/organizations/${request.id}`,
    );
  }

  async ListOrganization(
    request: PagingRequest,
  ): Promise<ListOrganizationResponse> {
    return await requestClient.get<ListOrganizationResponse>('/organizations', {
      params: request,
    });
  }

  async UpdateOrganization(request: UpdateOrganizationRequest): Promise<Empty> {
    const id = request.data?.id;
    if (request.data !== null && request.data !== undefined) {
      request.data.id = undefined;
    }
    return await requestClient.put<Empty>(`/organizations/${id}`, request);
  }
}

export const defOrganizationService = new OrganizationServiceImpl();

File: backend/app/admin/service/internal/service/organization.go
package service

import (
	"context"
	"kratos-admin/pkg/middleware/auth"

	"github.com/go-kratos/kratos/v2/log"
	pagination "github.com/tx7do/kratos-bootstrap/api/gen/go/pagination/v1"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-admin/app/admin/service/internal/data"

	adminV1 "kratos-admin/api/gen/go/admin/service/v1"
	userV1 "kratos-admin/api/gen/go/user/service/v1"
)

type OrganizationService struct {
	adminV1.OrganizationServiceHTTPServer

	log *log.Helper

	uc *data.OrganizationRepo
}

func NewOrganizationService(uc *data.OrganizationRepo, logger log.Logger) *OrganizationService {
	l := log.NewHelper(log.With(logger, "module", "organization/service/admin-service"))
	return &OrganizationService{
		log: l,
		uc:  uc,
	}
}

func (s *OrganizationService) List(ctx context.Context, req *pagination.PagingRequest) (*userV1.ListOrganizationResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *OrganizationService) Get(ctx context.Context, req *userV1.GetOrganizationRequest) (*userV1.Organization, error) {
	return s.uc.Get(ctx, req)
}

func (s *OrganizationService) Create(ctx context.Context, req *userV1.CreateOrganizationRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("错误的参数")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	if err = s.uc.Create(ctx, req, operator); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *OrganizationService) Update(ctx context.Context, req *userV1.UpdateOrganizationRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("错误的参数")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	if err = s.uc.Update(ctx, req, operator); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *OrganizationService) Delete(ctx context.Context, req *userV1.DeleteOrganizationRequest) (*emptypb.Empty, error) {
	if _, err := s.uc.Delete(ctx, req); err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}
